pkg/test/service/echo/application.go

```diff
@@ -0,0 +1,201 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package echo
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/test/service/echo/proto"
+)
+
+// Application is a simple application than processes echo requetss via various transports.
+type Application struct {
+	// Ports are the ports that the application should listen on. If any port number is 0, an available port will be selected
+	// when the application is started.
+	Ports model.PortList
+	// TLSCert defines the server-side TLS cert to use with GRPC.
+	TLSCert string
+	// TLSKey defines the server-side TLS key to use with GRPC.
+	TLSCKey string
+	// Version string
+	Version string
+
+	servers []serverInterface
+}
+
+// Start starts this application.
```

Comment:
Minor nit: Can shorten to "Start the application.'"

Reply:
done.